blockchain/chain.go

```diff
@@ -665,6 +670,37 @@ func (b *BlockChain) connectBlock(node *blockNode, block *btcutil.Block,
 	b.sendNotification(NTBlockConnected, block)
 	b.chainLock.Lock()
 
+	// If the pruneTarget isn't 0, we should attempt to delete older blocks
+	// from the database.
+	if b.pruneTarget != 0 {
+		err = b.db.Update(func(dbTx database.Tx) error {
+			// When the total block size is under the prune target, prune blocks is
+			// a no-op and the deleted hashes are nil.
+			deletedHashes, err := dbTx.PruneBlocks(b.pruneTarget)
+			if err != nil {
+				return err
+			}
+
+			// No need to try to delete the spend journal entries or the indexes
+			// if we haven't deleted any blocks.
+			if deletedHashes == nil {
```

Comment:
Check `len()` instead of `== nil` to ignore underlying implementation assumptions?

Reply:
Addressed in the latest push